client/client.go

```diff
@@ -303,8 +303,9 @@ func (r *NotaryRepository) RemoveTarget(targetName string) error {
 	return nil
 }
 
-// ListTargets lists all targets for the current repository
-func (r *NotaryRepository) ListTargets() ([]*Target, error) {
+// ListTargets lists all targets for the current repository. The list of
+// roles should be passed in order from lowest to highest priority
+func (r *NotaryRepository) ListTargets(roles ...string) ([]*Target, error) {
```

Comment:
Non-blocking question - is there a case where we want to pass no roles?  Maybe just to trigger an update from the server?